modules/worker/pool.go

```diff
@@ -45,115 +45,128 @@ func NewWorkerPool(queueSize int, workerCount int) *Pool {
 		Msg("Creating new worker pool")
 
 	p := &Pool{
+		logger:      logger,
 		workerCount: workerCount,
 		queue:       make(chan modules.Job, queueSize),
-		logger:      logger,
 	}
 
 	p.Manager = component.NewManager(
+		logger,
 		component.WithStartupLogic(func(ctx modules.ThrowableContext) {
-			// Startup logic - store context
-			p.ctx = ctx
-			p.logger.Trace().Msg("Starting worker pool")
-			// Start all workers
-			for i := 0; i < p.workerCount; i++ {
-				p.wg.Add(1)
-				workerID := i
-				p.logger.Trace().Int("worker_id", workerID).Msg("Starting worker")
-				go p.runWorker(workerID)
-			}
-			// Signal ready immediately after workers start
-			p.logger.Trace().Msg("All workers started, startup complete")
+			p.startWorkers(ctx)
 		}),
 		component.WithShutdownLogic(func() {
-			p.logger.Trace().Msg("initiating shutdown")
-
-			// Close queue to signal workers to stop
-			p.logger.Trace().Msg("Closing job queue")
-			close(p.queue)
-
-			// Wait for all workers to finish processing
-			p.logger.Trace().Msg("Waiting for all workers to finish")
-			p.wg.Wait()
-
-			// Signal done after all workers have finished
-			p.logger.Trace().Msg("All workers finished, shutdown complete")
+			p.stopWorkers()
 		}),
 	)
 
 	return p
 }
 
-// runWorker executes jobs from the queue until shutdown.
+// Submit adds a job to the worker pool queue.
+// Returns error if queue is full or pool has been shut down.
 // Args:
-//   - workerID: unique identifier for logging
+//   - job: the job to execute
 //
-// Exits on context cancellation or queue closure.
-func (p *Pool) runWorker(workerID int) {
-	defer func() {
+// Returns error if job cannot be submitted.
+func (p *Pool) Submit(job modules.Job) error {
+	p.logger.Trace().
+		Msg("Job submitted to pool")
+
+	select {
+	case <-p.ctx.Done():
```

Comment:
The `ctx` field is set in `Start()` but accessed in `Submit()` without synchronization. If `Submit()` is called before `Start()`, this will panic with nil pointer dereference. I'd suggest checking whether the `pool` is started inside the `Submit` like below, or use `p.Ready()`
```suggestion
      if p.ctx == nil {
          return fmt.Errorf("pool not started")
      }
	select {
	case <-p.ctx.Done():
```